### PR TITLE
feat(pipeline executions/orca): Add support for explicit opt-out from…

### PR DESCRIPTION
**Pull request by titirigaiulian**
… passing outputs to child executions

This PR introduces a feature flag `skipDownstreamOutput` that can be defined in `Run Pipeline` stage. By enabling it, clears the output of the parent pipeline, without affecting the child pipeline. 
This helps for stability issues when there are large deployment pipelines with multiple levels of parents. It can be set any parent level of the hierarchy.

Issue: https://github.com/spinnaker/spinnaker/issues/6159

### CODE REVIEW

**Comment by anthonywoo** [comment]
+1 for this option. We have a pipeline orchestrating other pipelines which results in huge writes to our RDS database

 What do we need to do to get this feature pushed forward?

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java [hunk 1]
```diff
@@ -71,6 +72,17 @@ public void taskGraph(@Nonnull StageExecution stage, @Nonnull TaskNode.Builder b
     }
   }
 
+  @Override
+  public void afterStages(@Nonnull StageExecution parent, @Nonnull StageGraphBuilder graph) {
+    if (parent
```
Comment (costimuraru) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java:77]: Nitpick: to make the `if` statement easier to understand, what do you think about extracting the condition in a dedicated method?
Something like:


```java
  @Override
  public void afterStages(@Nonnull StageExecution parent, @Nonnull StageGraphBuilder graph) {
    if (shouldSkipDownstreamOutput(parent)) {
      parent.setOutputs(emptyMap());
    }
  }

  private void shouldSkipDownstreamOutput(StageExecution parent) {
    return parent
        .getContext()
        .getOrDefault("skipDownstreamOutput", "false")
        .toString()
        .equals("true")
  }
```

**Comment by Nirmalyasen** [comment]
spinnaker/spinnaker#6159 is affecting quite a few of us. This fix does reduce the payload and does not force anything on everybody. Users can decide if they want the outputs propagated or not. Can this be accepted to help Spinnaker scale?

**Comment by emagana** [comment]
We are facing the same issue and this looks like a clean and easy fix. Maybe just waiting for @titirigaiulian  to make the changes suggested in the code review. Thanks!

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java [hunk 2]
```diff
@@ -71,6 +72,17 @@ public void taskGraph(@Nonnull StageExecution stage, @Nonnull TaskNode.Builder b
     }
   }
 
+  @Override
+  public void afterStages(@Nonnull StageExecution parent, @Nonnull StageGraphBuilder graph) {
```
Comment (Nirmalyasen) [inline orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java:76]: That parameter parent is really the current stage itself and not the parent. So, no calling it parent will reduce some confusion while reading the code.

Comment (titirigaiulian) [reply orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java]: done

File: orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java (hunk 1 above)
Comment (titirigaiulian) [reply orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/pipeline/PipelineStage.java]: done

File: orca-clouddriver/src/test/kotlin/com/netflix/spinnaker/orca/clouddriver/pipeline/KubernetesPreconfiguredJobSpec.kt [hunk 3]
```diff
@@ -85,7 +85,7 @@ class KubernetesPreconfiguredJobSpec : JUnit5Minutests {
             .contains("\"ref\":\"/pipelines")
         }
 
-        verify(timeout = 1500) { katoRestService.requestOperations(any(), "kubernetes", match { it.toString().contains("alias=preconfiguredJob") }) }
+        verify(timeout = 2000) { katoRestService.requestOperations(any(), "kubernetes", match { it.toString().contains("alias=preconfiguredJob") }) }
```
Comment (titirigaiulian) [inline orca-clouddriver/src/test/kotlin/com/netflix/spinnaker/orca/clouddriver/pipeline/KubernetesPreconfiguredJobSpec.kt:88]: this caused some build failures

**Comment by anthonywoo** [comment]
Any updates on this?

We've been using our own version of orca (`1.22.2 `) which contains this PR. We haven't had any issues and our pipelines page is more stable now

**Comment by dbrougham** [comment]
@costimuraru - Just wondering if there is anything outstanding that is preventing this from being merged in? We have noticed a similar issue in our prod Spinnaker instance with nested pipelines

**Review by robzienert — APPROVED** [review]
Sorry I missed this!

**Comment by dbrougham** [comment]
Thanks @robzienert for taking a look 🙌

**Pull request merged**